src/main/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfiguration.java

```diff
@@ -193,5 +245,86 @@ public String getParquetFileFormatName() {
     public void setParquetFileFormatName(String parquetFileFormatName) {
       this.parquetFileFormatName = parquetFileFormatName;
     }
+
+    public String getEncryptionKey() {
+      return encryptionKey;
+    }
+
+    public void setEncryptionKey(String encryptionKey) {
+      this.encryptionKey = encryptionKey;
+    }
+
+    public boolean isInitialize() {
+      return initialize;
+    }
+
+    public void setInitialize(boolean initialize) {
+      this.initialize = initialize;
+    }
+  }
+
+  /** Track the monitoring-related configuration */
+  public static class Monitoring {
+    // The resource ID of the Azure Logic app that handles sending Slack notifications
+    private String notificationApplicationId;
+    // The list of regional storage accounts to send long term logs to
+    private List<LogCollectionConfig> logCollectionConfigs;
+
+    public String getNotificationApplicationId() {
+      return notificationApplicationId;
+    }
+
+    public void setNotificationApplicationId(String notificationApplicationId) {
+      this.notificationApplicationId = notificationApplicationId;
+    }
+
+    public List<LogCollectionConfig> getLogCollectionConfigs() {
+      return logCollectionConfigs;
+    }
+
+    public void setLogCollectionConfigs(List<LogCollectionConfig> logCollectionConfigs) {
+      this.logCollectionConfigs = logCollectionConfigs;
+    }
+
+    public Map<AzureRegion, String> getLogCollectionConfigsAsMap() {
+      return logCollectionConfigs.stream()
+          .collect(
+              Collectors.toMap(
+                  LogCollectionConfig::getRegion,
+                  LogCollectionConfig::getTargetStorageAccountResourceId));
+    }
+  }
```

Comment:
Maybe not for this PR, but I think most configs can (and should) be migrated to use `record`. For example, this how you'd write this using `record`:
```suggestion
  /**
   * Track the monitoring-related configuration
   * @param notificationApplicationId The resource ID of the Azure Logic app that handles sending Slack notifications
   * @param logCollectionConfigs The list of regional storage accounts to send long term logs to
   */
  public record Monitoring(String notificationApplicationId, List<LogCollectionConfig> logCollectionConfigs) {
    public Map<AzureRegion, String> getLogCollectionConfigsAsMap() {
      return logCollectionConfigs.stream()
          .collect(
              Collectors.toMap(
                  LogCollectionConfig::getRegion,
                  LogCollectionConfig::getTargetStorageAccountResourceId));
    }
  }
```

Reply:
Ah, that's a great call.  I've created a ticket to just convert all config into records
https://broadworkbench.atlassian.net/browse/DR-3128